process.c

```diff
@@ -3011,6 +3012,31 @@ rb_execarg_fail(VALUE execarg_obj, int err, const char *errmsg)
 }
 #endif
 
+/*
+ *  call-seq:
+ *     Process.argv    -> array
+ *
+ *  Returns the list of original command line arguments passed to the
+ *  Ruby executable, while +ARGV+ are the arguments for the Ruby program.
+ *
+ *  Example:
+ *
+ *      $ ruby -e 'p Process.argv'
+ *      ["ruby", "-e", "p Process.argv"]
```

Comment:
I wonder if we should include `argv[0]`.
My main concern is it's a mistake to use that to reexecute, `argv[0]` can be set to anything, `RbConfig.ruby`/`/proc/self/exe`/etc is more reliable.
So I think we should exclude it to avoid that mistake.

I feel it's also more useful to separate Ruby flags from program arguments (there are none in this example, would be clearer if there are some what's the behavior about them), because one might want to use the same flags but run a different script/file.

Reply:
In other words, for `ruby -e 'p Process.argv'` I think the method should return `[]`
and for e.g `ruby -v --yjit -e 'p Process.argv'` it should return `["-v", "--yjit"]`.
So like `vm_args` in the original issue description.